cloud/shared/bin/lib/config_loader.py

```diff
@@ -214,50 +215,52 @@ def _validate_infra_variables(
 
         return validation_errors
 
-    def _get_commit_sha_for_release(self, tag: str) -> str:
+    def _get_commit_sha_for_tag(self, tag: str) -> str:
         """Get the commit SHA for the release specified in the tag.
         
-          The tag is a release version number such as "v1.24.0".
+          The tag can be a release version number such as "v1.24.0", a specific docker snapshot 
+          tag such as "SNAPSHOT-920bc49-1685642238" (where the middle secion is the shortened 
+          sha for the commit the docker image was built from), or the string "latest".
 
           We are calling the GitHub API with unauthenticated requests, which are rate-limited.
           The rate limit allows for up to 60 requests per hour associated with the originating 
           IP address.
         """
-        tag = tag.strip()
-        if tag == 'latest':
-            # Translate "latest" into a version number
-            tag = self._get_latest_version_number()
 
-        release_url = f"https://api.github.com/repos/civiform/civiform/git/refs/tags/{tag}"
-        release_response = requests.get(release_url)
+        tag = tag.strip()
 
-        if release_response.status_code == 200:
-            return self._get_commit_sha_for_tag(
-                release_response.json()["object"]["sha"])
-        else:
-            raise self.VersionNotFoundError(
-                f"The commit sha for version {tag} could not be found. Are you using a valid tag such as latest or a valid version number like v1.0.0? {release_response.status_code} - {release_response.json()['message']}"
-            )
+        try:
+            if tag == "latest":
+                return self._fetch_json_val(
+                    "https://api.github.com/repos/civiform/civiform/branches/main",
+                    "commit", "sha")
+            elif "SNAPSHOT" in tag:
+                short_sha = tag.split("-")[1]
+                return self._fetch_json_val(
+                    f"https://api.github.com/repos/civiform/civiform/commits/{short_sha}",
+                    "sha")
+            else:
+                tag_url = self._fetch_json_val(
+                    f"https://api.github.com/repos/civiform/civiform/git/refs/tags/{tag}",
+                    "object", "url")
+                return self._fetch_json_val(tag_url, "object", "sha")
```

Comment:
**suggestion (blocking):** please log which of these branches is executes and intermediate values such as `tag_url` on L243 so that if any of them fails we know which it was

Reply:
done